libsolidity/ast/Types.cpp

```diff
@@ -2074,7 +2074,11 @@ std::unique_ptr<ReferenceType> ArrayType::copyForLocation(DataLocation _location
 
 BoolResult ArraySliceType::isImplicitlyConvertibleTo(Type const& _other) const
 {
-	if (m_arrayType.location() == DataLocation::CallData && m_arrayType.isDynamicallySized() && m_arrayType == _other)
+	if (
+		m_arrayType.dataStoredIn(DataLocation::CallData) &&
+		m_arrayType.isDynamicallySized() &&
+		m_arrayType.isImplicitlyConvertibleTo(_other)
+	)
 		return true;
 	return (*this) == _other;
```

Comment:
Since these are all simple boolean checks, how about this style?

```suggestion
	return
		(*this) == _other ||
		m_arrayType.dataStoredIn(DataLocation::CallData) &&
		m_arrayType.isDynamicallySized() &&
		m_arrayType.isImplicitlyConvertibleTo(_other);
```